mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -194,15 +192,18 @@ public void setCurrentKillstreaks(Transaction transaction, int amount) {
             throw new IllegalArgumentException("amount must be positive");
         }
 
-        transaction.getProperty(DSLContext.class).update(KITPVP_STATISTICS)
-                .set(KITPVP_STATISTICS.HIGHEST_KILLSTREAK,KITPVP_STATISTICS.HIGHEST_KILLSTREAK.greatest(amount))
-                .set(KITPVP_STATISTICS.CURRENT_KILLSTREAK,amount)
-                .execute();
 
-        greatest(KITPVP_STATISTICS.HIGHEST_KILLSTREAK)
+        KitpvpStatisticsRecord record = getStatistics(transaction);
+
+        int highest = Math.max(record.getHighestKillstreak(),amount);
 
+        record.setHighestKillstreak(highest);
+        record.setCurrentKillstreak(amount);
+
+        record.store(KITPVP_STATISTICS.HIGHEST_KILLSTREAK,KITPVP_STATISTICS.CURRENT_KILLSTREAK);
 
         this.updateCurrentKillstreak(amount);
+        this.updateHighestKillstreak(amount);
```

Comment:
Should be `updateHighestKillstreak(highest)`

Reply:
I was actually wrong a248, this would be better to leave separate from update highest killstreak. The only method we need is one that resets the current killstreak, and that is all. I will be updating the PR later tonight.